mobile/src/main/java/rs/readahead/washington/mobile/domain/usecases/shared/ScheduleUploadReportFilesUseCase.kt

```diff
@@ -6,43 +8,52 @@ import androidx.work.OneTimeWorkRequest
 import androidx.work.WorkManager
 import com.google.firebase.crashlytics.FirebaseCrashlytics
 import com.hzontal.tella_vault.VaultFile
+import dagger.hilt.android.qualifiers.ApplicationContext
 import io.reactivex.Completable
-import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.schedulers.Schedulers
+import org.hzontal.tella.keys.key.LifecycleMainKey
+import rs.readahead.washington.mobile.MyApplication
 import rs.readahead.washington.mobile.data.database.KeyDataSource
 import rs.readahead.washington.mobile.domain.entity.collect.FormMediaFile
-import rs.readahead.washington.mobile.domain.usecases.base.CompletableUseCase
 import rs.readahead.washington.mobile.util.jobs.WorkerUploadReport
 import javax.inject.Inject
 
 class ScheduleUploadReportFilesUseCase @Inject constructor(
     private val keyDataSource: KeyDataSource,
-    private val vaultFile: VaultFile,
-    private val uploadServerId: Long,
+    @ApplicationContext
     private val context : Context
-) : CompletableUseCase() {
+) {
 
-    override fun buildUseCaseSingle(): Completable {
+    fun execute(vaultFile: VaultFile, uploadServerId: Long): Completable {
         return keyDataSource.dataSource
             .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
             .flatMapCompletable { dataSource ->
-                dataSource.scheduleUploadReport(FormMediaFile.fromMediaFile(vaultFile), uploadServerId)
-            }
-            .observeOn(AndroidSchedulers.mainThread())
-            .doOnComplete {
+                val formMediaFile = FormMediaFile.fromMediaFile(vaultFile)
                 val constraints = Constraints.Builder()
                     .setRequiredNetworkType(NetworkType.CONNECTED)
                     .build()
-                val onetimeJob = OneTimeWorkRequest.Builder(WorkerUploadReport::class.java)
+
+                val oneTimeJob = OneTimeWorkRequest.Builder(WorkerUploadReport::class.java)
                     .setConstraints(constraints)
                     .build()
-                WorkManager.getInstance(context)
-                    .enqueueUniqueWork("WorkerUploadReport2", ExistingWorkPolicy.KEEP, onetimeJob)
 
+                // Schedule the upload report work using WorkManager
+                WorkManager.getInstance(context).enqueueUniqueWork(
+                    "WorkerUploadReport2",
+                    ExistingWorkPolicy.KEEP,
+                    oneTimeJob
+                )
+
+                // Update the main key holder's timeout
+                MyApplication.getMainKeyHolder().timeout = LifecycleMainKey.NO_TIMEOUT
+
+                // Perform the upload report operation and complete the Completable
```

Comment:
Let's add a doFinally block to ensure that the work is cleared from WorkManager using "cancelUniqueWork" after the upload report operation is completed.